lib/matplotlib/tests/test_rcparams.py

```diff
@@ -422,3 +422,68 @@ def test_rcparams_reset_after_fail():
                 pass
 
         assert mpl.rcParams['text.usetex'] is False
+
+
+def test_if_rctemplate_is_up_to_date():
+    # This tests if the matplotlibrc.template file
+    # contains all valid rcParams.
+    dep1 = mpl._all_deprecated
+    dep2 = mpl._deprecated_set
+    deprecated = list(dep1.union(dep2))
+    #print(deprecated)
+    path_to_rc = mpl.matplotlib_fname()
+    with open(path_to_rc, "r") as f:
+        rclines = f.readlines()
+    missing = {}
+    for k,v in mpl.defaultParams.items():
+        if k[0] == "_":
+            continue
+        if k in deprecated:
+            continue
+        if "verbose" in k:
+            continue
+        found = False
+        for line in rclines:
+            if k in line:
+                found = True
+        if not found:
+            missing.update({k:v})
+    if missing:
+        raise ValueError("The following params are missing " +
+                         "in the matplotlibrc.template file: {}"
+                         .format(missing.items()))
+    assert False
+
+
+def test_if_rctemplate_would_be_valid():
+    # This tests if the matplotlibrc.template file would result in a valid
+    # rc file if all lines are uncommented.
+    path_to_rc = mpl.matplotlib_fname()
+    with open(path_to_rc, "r") as f:
+        rclines = f.readlines()
+    newlines = []
+    for line in rclines:
+        if line[0] == "#":
+            newline = line[1:]
+        else:
+            newline = line
+        if "$TEMPLATE_BACKEND" in newline:
+            newline = "backend : Agg"
+        if "datapath" in newline:
+            newline = ""
+        newlines.append(newline)
+    #print(os.path.dirname(__file__))
+    fname = os.path.join(os.path.dirname(__file__),
+                         'testrcvalid.temp')
```

Comment:
pytest has a tempfile fixture that would be better to use for this purpose.

Reply:
What exactly is the problem with that? Do I not have permission to create a file? Using a fixture to create a temporary directory and create a file inside seems rather overkill.